src/main/java/plugily/projects/buildbattle/arena/managers/ScoreboardManager.java

```diff
@@ -50,29 +48,34 @@ public ScoreboardManager(PluginArena arena) {
+    PluginArena userArena = user.getArena();
+
+    if(userArena.getArenaState() == ArenaState.FULL_GAME) {
+      lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content.Starting");
+    } else if(userArena.getArenaState() == ArenaState.IN_GAME) {
+      if(userArena instanceof GuessArena) {
+        lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + userArena.getArenaState().getFormattedName() + ".Guess-The-Build" + (((GuessArena) userArena).getArenaInGameStage() == BaseArena.ArenaInGameStage.PLOT_VOTING ? "-Waiting" : ""));
       } else {
-        if(user.getArena().getArenaOption("PLOT_MEMBER_SIZE") <= 1) {
-          lines = user.getArena().getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + user.getArena().getArenaState().getFormattedName() + ".Classic");
+        if(userArena.getArenaOption("PLOT_MEMBER_SIZE") <= 1) {
+          lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + userArena.getArenaState().getFormattedName() + ".Classic");
         } else {
-          lines = user.getArena().getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + user.getArena().getArenaState().getFormattedName() + ".Teams");
+          lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + userArena.getArenaState().getFormattedName() + ".Teams");
         }
       }
-    } else if(user.getArena().getArenaState() == ArenaState.ENDING) {
-      if(user.getArena() instanceof GuessArena) {
-        lines = user.getArena().getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + user.getArena().getArenaState().getFormattedName() + ".Guess-The-Build");
+    } else if(userArena.getArenaState() == ArenaState.ENDING) {
+      if(userArena instanceof GuessArena) {
+        lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + userArena.getArenaState().getFormattedName() + ".Guess-The-Build");
       } else {
-        lines = user.getArena().getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + user.getArena().getArenaState().getFormattedName() + ".Classic");
+        lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + userArena.getArenaState().getFormattedName() + ".Classic");
       }
     } else {
-      lines = user.getArena().getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + user.getArena().getArenaState().getFormattedName());
+      lines = userArena.getPlugin().getLanguageManager().getLanguageList("Scoreboard.Content." + userArena.getArenaState().getFormattedName());
     }
+
+    org.bukkit.entity.Player player = user.getPlayer();
```

Comment:
why not importing Player?

Reply:
and why defining it if its only used one time?

Reply:
> why not importing Player?

Using fully-qualified import for this is pretty useless as it is used once. If there will be more fields created like this we will import this Player.

> and why defining it if its only used one time?

`user#getPlayer` method invokes `Map#get` which pressures the CPU well on long distance, I mean if a game is running for more than 5 minutes, there are plenty of players joined to this game and 15 or more lines of string added to this scoreboard.

Reply:
- About import, I would like to see it imported as it looks better and cleaner
- did not saw for last time